src/Grammar.js

```diff
@@ -182,7 +177,8 @@ Grammar.prototype = {
       }
 
       var description = null;
-      if (!self.superGrammar.ruleBodies[ruleName] && self.ruleDescriptions[ruleName]) {
+      if (!(self.superGrammar && self.superGrammar.ruleBodies[ruleName]) &&
```

Comment:
... referencing it down here, i.e., `!overridingOrExtending && ...`? I think that would make this code a little more readable. Thanks!